test/cli/run/run-eval.test.ts

```diff
@@ -40,4 +40,12 @@ describe("bun -e", () => {
     expect(stderr.toString("utf8")).toInclude('"hi" as 2');
     expect(stderr.toString("utf8")).toInclude("Unexpected throw");
   });
+
+  test("positional argv is not mangled", async () => {
```

Comment:
I'm skeptical this will fix the issue without leaking other parameters into it

but we shall see